### PR TITLE
Disable threading on browser

### PR DESCRIPTION
**Pull request by lambdageek**
Fixes https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/690

### CODE REVIEW

**Comment by lambdageek** [comment]
@KirillOsenkov @yuehuang010 Do you want PRs against `main` or `live`?  I would prefer to push this to `main` (I'd like to use `StructuredLogger` on another platform that doesn't have threads)

**Comment by KirillOsenkov** [comment]
Let’s target main and I’ll merge into live. Thanks!!

File: src/StructuredLogger/PlatformUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace StructuredLogger;
+
+public class PlatformUtilities
+{
+    public static bool HasThreads => !RuntimeInformation.IsOSPlatform(_browser);
```
Comment (yuehuang010) [inline src/StructuredLogger/PlatformUtilities.cs:8]: Is the function IsOSPlatform fast?  Can we cache the result since the value won't change?

Comment (lambdageek) [reply src/StructuredLogger/PlatformUtilities.cs]: Done

**Comment by lambdageek** [comment]
Rebased the PR on `main`

**Comment by lambdageek** [comment]
This is ready to merge, @KirillOsenkov @yuehuang010

**Review by KirillOsenkov — APPROVED** [review]
Thanks for doing this! 

I was procrastinating on getting the WASM version working again after we added parallelism and it is a huge relief that you did it for us.

**Comment by KirillOsenkov** [comment]
Just fixed up the namespace a little bit:
https://github.com/KirillOsenkov/MSBuildStructuredLog/commit/cfd1aa97b16d49bbe05118e450c2460310cb74fa

**Comment by KirillOsenkov** [comment]
I've published the latest version to https://live.msbuildlog.com, thanks!

**Pull request merged**